File: server/src/main/java/cn/iinti/atom/entity/metric/Metric.java
package cn.iinti.atom.entity.metric;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import io.micrometer.core.instrument.Meter;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.experimental.Accessors;

import java.io.Serializable;
import java.time.LocalDateTime;


/**
 * <p>
 * 监控指标
 * </p>
 *
 * @author iinti
 * @since 2023-03-13
 */
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
@Schema(name = "Metric对象", description = "监控指标")
public class Metric implements Serializable {

    private static final long serialVersionUID = 1L;

    @Schema(name = "自增主建")
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    @Schema(name = "指标名称")
    private String name;

    @Schema(name = "时间索引")
    private String timeKey;


    @Schema(name = "对于tag字段自然顺序拼接求md5")
    private String tagsMd5;

    @Schema(name = "指标tag")
    private String tag1;

    @Schema(name = "指标tag")
    private String tag2;

    @Schema(name = "指标tag")
    private String tag3;

    @Schema(name = "指标tag")
    private String tag4;

    @Schema(name = "指标tag")
    private String tag5;

    @Schema(name = "指标类型：（counter、gauge、timer，请注意暂时只支持这三种指标）")
    private Meter.Type type;

    @Schema(name = "指标值")
    private Double value;

    @Schema(name = "创建时间")
    private LocalDateTime createTime;


    public static final String ID = "id";

    public static final String NAME = "name";

    public static final String TIME_KEY = "time_key";

    public static final String TAGS_MD5 = "tags_md5";

    public static final String TAG1 = "tag1";

    public static final String TAG2 = "tag2";

    public static final String TAG3 = "tag3";

    public static final String TAG4 = "tag4";

    public static final String TAG5 = "tag5";

    public static final String TYPE = "type";

    public static final String VALUE = "value";

    public static final String CREATE_TIME = "create_time";

}

File: server/src/main/kotlin/cn/iinti/katom/entity/metric/Metric.kt
package cn.iinti.katom.entity.metric

import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableId
import io.micrometer.core.instrument.Meter
import io.swagger.v3.oas.annotations.media.Schema
import lombok.Data
import lombok.EqualsAndHashCode
import lombok.experimental.Accessors
import java.io.Serializable
import java.time.LocalDateTime


@Schema(name = "Metric对象", description = "监控指标")
@Data
@EqualsAndHashCode(callSuper = false)
@Accessors(chain = true)
open class Metric(
    @Schema(name = "自增主建")
    @TableId(value = "id", type = IdType.AUTO)
    var id: Long? = null,

    @Schema(name = "指标名称")
    var name: String? = null,

    @Schema(name = "时间索引")
    var timeKey: String? = null,

    @Schema(name = "对于tag字段自然顺序拼接求md5")
    var tagsMd5: String? = null,

    @Schema(name = "指标tag")
    var tag1: String? = null,

    @Schema(name = "指标tag")
    var tag2: String? = null,

    @Schema(name = "指标tag")
    var tag3: String? = null,

    @Schema(name = "指标tag")
    var tag4: String? = null,

    @Schema(name = "指标tag")
    var tag5: String? = null,

    @Schema(name = "指标类型：（counter、gauge、timer，请注意暂时只支持这三种指标）")
    var type: Meter.Type? = null,

    @Schema(name = "指标值")
    var value: Double? = null,

    @Schema(name = "创建时间")
    var createTime: LocalDateTime? = null
) : Serializable {

    companion object {
        const val ID = "id"
        const val NAME = "name"
        const val TIME_KEY = "time_key"
        const val TAGS_MD5 = "tags_md5"
        const val TAG1 = "tag1"
        const val TAG2 = "tag2"
        const val TAG3 = "tag3"
        const val TAG4 = "tag4"
        const val TAG5 = "tag5"
        const val TYPE = "type"
        const val VALUE = "value"
        const val CREATE_TIME = "create_time"
    }
}